src/quotes.c

```diff
@@ -53,42 +55,41 @@ int	has_quotes(char *str, char *q)
  * And then it constructs the final string in 2 steps, first joining part_1 and
  * part_2 into tmp_join and then joining tmp_join and part_3 into final.
  * 
- * Finally, the appropriate string in the 2D input array is pointed to the final
- * string. All of the dynamically allocated memory is then freed with the
- * exception of the final string that is not freed until after the whole command
- * is processed and the program reaches the free_simple_cmds function.
+ * All of the dynamically allocated memory is then freed with the exception
+ * of the final string that is returned. The final string is not freed until
+ * after the whole command is processed and the program reaches
+ * the free_simple_cmds function.
  * 
- * @param 	cmd		array of strings containing the input commands
- * @param	i_cmd	index of the command string in the array to be modified
+ * @param 	str		string to check and potentially remove the quotes from
```

Comment:
Changed the function so if there are quotes to delete it does so "in_place", meaning that it uses the old str to extract the new one, frees the old one and returns the new one.